common/constants.go

```diff
@@ -54,6 +54,8 @@ const (
 	LastBlobNextPageToken = -1
 	// EndMessageID is the id of the end message, here we use the int64 max
 	EndMessageID int64 = 1<<63 - 1
+	// FirstWorkflowTaskScheduledEventID is the id of the first WorkflowTaskScheduled event in a workflow
+	FirstWorkflowTaskScheduledEventID int64 = FirstEventID + 1
```

Comment:
That's not necessarily true though, there could be signals before the first WFT schedule event

Reply:
Ah, okay. So we do need to pass this from history, you think?

Reply:
I don't think you can avoid providing the information from history service.

Reply:
you're right. I think I found a way to do it without adding even more stuff to mutable state